docs/source/building/shared-libraries.rst

```diff
@@ -0,0 +1,38 @@
+
+To use relative links, we need to use a special variable in the link itself.  In Linux, the $ORIGIN variable allows you to say "relative to this file as it's being executed".  On OS X, there are ``@rpath``, ``@loader_path``, ``@executable_path``.  ``@rpath`` allows you to set relative links from the system load paths, while ``@loader_path`` is equivalent to ``$ORIGIN``, and ``@executable_path`` supports the Apple ".app" directory approach, where libraries know where they live relative to their calling application. 
+
+Conda build uses ``$ORIGIN`` on Linux and ``@loader_path`` on Mac OS X since we install into a common root directory and can assume that other libraries will also be installed into that root.  The use of the variables allows us to build *relocatable binaries* that can be built on one person's system and sent everywhere.
+
+On Linux, ``conda-build`` modifies any shared libraries or generated executables to use a relative dynamic link by calling the ``patchelf`` tool. On Mac OS X, the ``install_name_tool`` tool is used.
+
+CAUTION: setting ``LD_LIBRARY_PATH`` (on Linux) or ``DYLD_LIBRARY_PATH`` (on Mac OS X) can interfere with this because the dynamic linker short-circuits link resolution by first looking at ``LD_LIBRARY_PATH``. 
+
+For example, let's assume that you decided to install an old version of ``libcurl`` into your conda environment due to some compatibility issues with the code you're using.  Then, you set ``export LD_LIBRARY_PATH=/home/jsmith/envs/curl_env/lib``.  From now on, every program you execute in that session will favor this libcurl to your system libcurl, since it's now effectively at the "front" of the dynamic load path. 
+
+Therefore, we discourage the inclusion of conda environment paths in ``LD_LIBRARY_PATH`` or ``DYLD_LIBRARY_PATH``.
+
+Shared libraries in Windows
+---------------------------
+
+Windows does not have the same concept of embedding links into binaries. Instead, Windows depends primarily on searching directories for matching filenames, as documented at https://msdn.microsoft.com/en-us/library/7d83bc18.aspx. 
+
+There is an alternate configuration, called `side-by-side assemblies <https://en.wikipedia.org/wiki/Side-by-side_assembly>`_, that requires specification of DLL versions in either an embedded manifest, or an appropriately named xml file alongside the binary in question. Conda does not currently use side-by-side assemblies, but may turn towards that in the future to resolve complications with multiple versions of the same library on one system.
+
+For now, most DLLs are installed into (install prefix)\Library\bin. As of 2015/11/09, this path is added to os.environ["PATH"] for all Python processes, so that DLLs can be located, regardless of the value of the system's PATH environment variable.
```

Comment:
the backslashes need to be escaped or I believe the text will not render properly. So I think the text should say:

```
(install prefix)\\Library\\bin.
```

I am not 100% certain because I don't know how this file is processed, but it is true that the backslashes don't render when you click the "View" button for the file.